src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Util.daml

```diff
@@ -544,6 +547,36 @@ calculateFloatingPaymentClaimsFromSwapStream floatingRateCalculation s
             where
               rate = getFloatingRate s pp.calculationPeriod p regularRate rateFixingCalendar fixLeg
 
+      -- calculate simple interest rate claim (no calc period compounding) for a payment period
+      -- using a compounded reference rate
+      createSimpleClaimCompoundedRefRate regularRate p =
+        mconcat $ map createSimpleRateCompoundingRefRate paymentPeriodData
+        where
+          (notional, paymentPeriodData) = calculateNotionalAndFixings p s
+            calculationScheduleStaticData fxRateId useAdjustedDatesForDcf periodicSchedule
+          createSimpleRateCompoundingRefRate pp = mconcat $ [claim, correctionClaim]
+            where
+              rate = getFloatingRate s pp.calculationPeriod p regularRate rateFixingCalendar fixLeg
+                -- Act360 was implicitly used for the SOFR Index, see page 35 here:
+                -- https://www.newyorkfed.org/medialibrary/Microsites/arrc/files/2021/users-guide-to-sofr2021-update.pdf
+              dcfMethodImpliedInSofrIndex = Act360
+              dcfImpl = Const (calcPeriodDcf dcfMethodImpliedInSofrIndex
+                pp.calculationPeriod useAdjustedDatesForDcf periodicSchedule.terminationDate
+                periodicSchedule.frequency)
+              notionalAndDcf = if pp.rateFixingDate < pp.calculationPeriod.adjustedEndDate then
+                error "fixing of notional must not occur before the final calculation date"
+                else
+                  when (TimeGte pp.rateFixingDate) $ scale notional $
+                  scale (pp.dcf / dcfImpl) $
+                  when (TimeGte p.adjustedEndDate) $ one currency
+              claim =
+                -- calculate compounded SOFR via the SOFR Index, see page 36 here:
+                -- https://www.newyorkfed.org/medialibrary/Microsites/arrc/files/2021/users-guide-to-sofr2021-update.pdf
+                when (TimeGte pp.calculationPeriod.adjustedStartDate) $ scale (Const 1.0 / rate) $
```

Comment:
Also, we can probably introduce a utility function in `Daml.Finance.Claims.Builders` (or even `ContingentClaims.Core.Builders`) to calculate a performance payoff, given that this is now used in quite a few places.